src/main/java/net/neoforged/moddevgradle/internal/ModDevRunWorkflow.java

```diff
@@ -0,0 +1,490 @@
+                versionCapabilites
+        );
+
+        project.getExtensions().add(EXTENSION_NAME, workflow);
+
+        return workflow;
+    }
+
+    public void configureTesting(Provider<ModModel> testedMod, Provider<Set<ModModel>> loadedMods) {
+        var testing = project.getExtensions().getByType(TestingExtension.class);
+        var testSuite = (JvmTestSuite) testing.getSuites().getByName("test");
+        var testSourceSet = testSuite.getSources();
+
+        var artifactsWorkflow = ModDevArtifactsWorkflow.get(project);
+        artifactsWorkflow.addToSourceSet(testSourceSet.getName());
+
+        var configurations = project.getConfigurations();
+
+        // If test fixtures are available for the current workflow, add them to runtime only
+        if (testFixturesDependency != null) {
+            configurations.getByName(testSourceSet.getRuntimeOnlyConfigurationName(), configuration -> {
+                configuration.getDependencies().add(testFixturesDependency);
+            });
+        }
+
+        for (var target : testSuite.getTargets()) {
+            setupTestTask(
+                    project,
+                    branding,
+                    userDevConfigOnly,
+                    target.getTestTask(),
+                    loadedMods,
+                    testedMod,
+                    artifactsWorkflow.modDevBuildDir(),
+                    modulePath -> {
+                        if (modulePathDependency != null) {
+                            modulePath.getDependencies().add(modulePathDependency);
+                        }
+                    },
+                    legacyClassPath -> {
+                        legacyClassPath.getDependencies().add(gameLibrariesDependency);
+                        addClientResources(project, legacyClassPath, artifactsWorkflow.createArtifacts());
+                    },
+                    artifactsWorkflow.downloadAssets().flatMap(DownloadAssets::getAssetPropertiesFile)
+            );
+        }
+    }
+
+    // FML searches for client resources on the legacy classpath
+    private void addClientResources(Project project, Configuration spec, TaskProvider<CreateMinecraftArtifacts> createArtifacts) {
+        // FML searches for client resources on the legacy classpath
+        spec.getDependencies().add(
+                project.getDependencyFactory().create(
+                        project.files(createArtifacts.flatMap(CreateMinecraftArtifacts::getResourcesArtifact))
+                )
+        );
+    }
+
+    public static void setupRuns(
+            Project project,
+            Branding branding,
+            Provider<Directory> argFileDir,
+            DomainObjectCollection<RunModel> runs,
+            Object runTemplatesSourceFile,
+            Consumer<Configuration> configureModulePath,
+            Consumer<Configuration> configureLegacyClasspath,
+            Provider<RegularFile> assetPropertiesFile,
+            Provider<VersionCapabilities> versionCapabilities
```

Comment:
Why is this a provider if it is always known ahead of time? In NeoDev you can supply null.